common/app/common/configuration.scala

```diff
@@ -57,37 +58,40 @@ object Environment extends Logging {
   */
 object GuardianConfiguration extends Logging {
 
-  import com.gu.cm.{Configuration => CM}
   import com.typesafe.config.Config
 
-  lazy val configuration = {
-    import Environment._
-
-    // This is version number of the config file we read from s3,
-    // increment this if you publish a new version of config
-    val s3ConfigVersion = 51
-
-    lazy val userPrivate = FileConfigurationSource(s"${System.getProperty("user.home")}/.gu/frontend.conf")
-    lazy val runtimeOnly = FileConfigurationSource("/etc/gu/frontend.conf")
-    lazy val identity = AwsApplication(stack, app, stage, awsRegion)
-    lazy val commonS3Config = S3ConfigurationSource(identity, configBucket, Configuration.aws.mandatoryCredentials, Some(s3ConfigVersion))
-    lazy val config = new CM(List(userPrivate, runtimeOnly, commonS3Config), SysOutLogger).load.resolve
-
-    // test mode is self contained and won't need to use anything secret
-    lazy val test = ClassPathConfigurationSource("env/DEVINFRA.properties")
-    lazy val testConfig = new CM(List(test), SysOutLogger).load.resolve
-
-    val appConfig =
-      if (stage == "DEVINFRA") testConfig
-      else {
-        try {
-          config.getConfig(identity.app + "." + identity.stage)
-        } catch {
-          case e: ConfigException if stage == "DEV" =>
-            throw new RuntimeException(s"${e.getMessage}.  You probably need to refresh your credentials.", e)
-        }
-      }
-    appConfig
+  private def configFromFile(path: String, configPath: String): Config = {
+    val fileConfig = ConfigFactory.parseFileAnySyntax(new File(path))
+    Try(fileConfig.getConfig(configPath)).getOrElse(ConfigFactory.empty)
+  }
+
+  private def configFromParameterStore(path: String): Config = {
+    val params = parameterStore.getPath(path)
+    val configMap = params.map {
+      case (key, value) => key.replaceFirst(s"$path/", "") -> value
+    }
+    ConfigFactory.parseMap(configMap.asJava)
+  }
+
+  private lazy val parameterStore = new ParameterStore(awsRegion)
+
+  lazy val configuration: Config = {
+    if (stage == "DEVINFRA")
+      ConfigFactory.parseResourcesAnySyntax("env/DEVINFRA.properties")
+    else {
+      val userPrivate = configFromFile(s"${System.getProperty("user.home")}/.gu/frontend.conf", "devOverrides")
+      val runtimeOnly =  configFromFile("/etc/gu/frontend.conf", "parameters")
+      val localConfig = userPrivate.withFallback(runtimeOnly)
```

Comment:
why an intermediary value here rather than doing the following below?
```
userPrivate
   withFallback(runtimeOnly)
   .withFallback(frontendAppConfig)
   .withFallback(frontendStageConfig)
   .withFallback(frontendConfig)
```

Reply:
No real reason, happy to change